gcloud/bigtable/column_family.py

```diff
@@ -15,6 +15,59 @@
 """User friendly container for Google Cloud Bigtable Column Family."""
 
 
+class GarbageCollectionRule(object):
+    """Table garbage collection rule.
+
+    Cells in the table fitting the rule will be deleted during
+    garbage collection.
+
+    .. note::
+
+        At most one of ``max_num_versions`` and ``max_age`` can be specified
+        at once.
```

Comment:
RE: "marshaling" Have a look at the `to_pb` methods in the linked to files.

I think it'd best to have a common ancestor so we can check for that ancestor when combining GC rules via unions and intersections (implemented in the downstream repo `gcloud-python-bigtable`) as well as combine filters via chains, interleaves and conditions (also implemented downstream).